includes/multi-currency/Compatibility/WooCommerceProductAddOns.php

```diff
@@ -230,8 +282,17 @@ public function update_product_price( $updated_prices, $cart_item, $prices ): ar
 	 */
 	public function order_line_item_meta( array $meta_data, array $addon, \WC_Order_Item_Product $item, array $values ): array {
 
+		$add_price_to_value = apply_filters( 'woocommerce_addons_add_order_price_to_value', false, $item );
+
+		$value = $addon['value'];
+
+		// Pass the timestamp as the add-on value in order to save the timestamp to the DB.
+		if ( isset( $addon['timestamp'] ) ) {
+			$value = $addon['timestamp'];
```

Comment:
Yes, this is 1-1 identical with what Product Add-Ons does. Since the WooPayments function overrides everything coming from Product Add-Ons, we should initialize the add-on value to the timestamp for Date Picker add-ons.